mono/metadata/w32socket-unix.c

```diff
@@ -320,11 +327,58 @@ mono_w32socket_recvfrom (SOCKET sock, char *buf, int len, int flags, struct sock
 
 	info = mono_thread_info_current ();
 
+	if (blocking) {
+		memset(&tstart, 0, timevallen);
```

Comment:
There's no risk in just zero-initializing these variables in their definitions (e.g. with `struct timeval tstart = {0}`).